docs/GettingStarted.md

```diff
@@ -2466,3 +2469,29 @@ However, additional instrumentation provided by Datadog can be activated alongsi
 | `OpenTracing::FORMAT_TEXT_MAP` | Yes        |                        |
 | `OpenTracing::FORMAT_RACK`     | Yes        | Because of the loss of resolution in the Rack format, please note that baggage items with names containing either upper case characters or `-` will be converted to lower case and `_` in a round-trip respectively. We recommend avoiding these characters or accommodating accordingly on the receiving end. |
 | `OpenTracing::FORMAT_BINARY`   | No         |                        |
+
+## Common issues
+
+### Payload too large
+
+By default, Datadog limits the size of trace payloads to prevent memory overhead within instrumented applications.  However, if your application creates long running traces, measured in minutes, or traces with thousands of operations, these traces may not be sent to Datadog due to these size limits.
+
+If traces are missing and to confirm this behavior is occurring, enable [debug mode](#tracer-settings) and check if messages similar to `"Dropping trace. Payload too large"` are being emitted.
+
+Debug mode is very verbose and it is not recommended to leave this enabled, so disable it after confirming. You can also inspect the [Datadog Agent logs](https://docs.datadoghq.com/agent/guide/agent-log-files/) for similar messages.
+
+Once you have confirmed traces are being dropped due to large payloads, enable the [partial_flush](#tracer-settings) setting as a tracer configuration. This will break down large traces into smaller chunks that get streamed to Datadog.
+
+### Stack level too deep
+
+`ddtrace` uses [Module#prepend](https://ruby-doc.org/core-3.0.0/Module.html#method-i-prepend) when instrumenting gems, as this approach allows for non-competing overriding of the same method by other libraries present in the same application.
+
+The presence of other libraries that use [alias](https://ruby-doc.org/core-3.0.0/doc/syntax/miscellaneous_rdoc.html#label-alias) or [Module#alias_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-alias_method) to override methods can cause issues to manifest in your application, often referred to as “noisy neighbor” issues.  The two methods above destructively replace the original method being overridden.  This approach is also incompatible with `Module#prepend` overriding, causing, for example, `SystemStackError`.
+
+A few libraries have known workarounds for these types of noisy neighbor issues:
+
+* `rack-mini-profiler`: [Net::HTTP stack level too deep errors](https://github.com/MiniProfiler/rack-mini-profiler#nethttp-stack-level-too-deep-errors).
+
+For libraries without a known workaround, consider removing the library using `alias` or `Module#alias_method`, separating libraries into different environments for testing, or creating a github issue with the affected libraries.
+
```

Comment:
I still kinda think we should ask the customers to report the issue to us first, as otherwise we may not notice that a customer is having issues, and we're the ones that can have more context to go to upstream, ask them to fix, etc.